### PR TITLE
Small improvements to allow usage of mkphyscloud outside of the SUSE environment

### PR DESCRIPTION
**Pull request by seife**
All these small enhancements are designed in a way that the old default behaviour is conserved, unless new configuration variables are set.

### CODE REVIEW

File: scripts/lib/mkcloud-driver-libvirt.sh [hunk 1]
```diff
@@ -374,8 +374,10 @@ function libvirt_do_onhost_deploy_image()
         fi
     fi
 
+    local qemu_img_extra=""
+    test -b $disk && qemu_img_extra="-t none -S 0" # sparse and cache only if it is not a blockdevice
     echo "Cloning $role node vdisk from $image ..."
-    safely $sudo qemu-img convert -t none -O raw -S 0 -p $image_path $disk
+    safely $sudo qemu-img convert -O raw $qemu_img_extra -p $image_path $disk
```
Comment (nicolasbock) [inline scripts/lib/mkcloud-driver-libvirt.sh]: Thanks!

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -1451,6 +1451,7 @@ function reboot_nodes_via_ipmi
     [[ $ipmi_ip_addrs ]] || set_crowbar_ipmi_ip_addrs
     local ip
     local i=0
+    : ${want_ipmi_pxeextraopts:=options=persistent}
```
Comment (nicolasbock) [inline scripts/qa_crowbarsetup.sh]: We typically define defaults at the beginning of this script. Could you move this definition to the others?

File: hostscripts/gatehost/freshadminvm [hunk 3]
```diff
@@ -7,7 +7,7 @@ if [ -z "$n" ] ; then
         exit 1
 fi
 
-if [[ "$n" != crowbard2 && "$n" != crowbard3 ]] ; then
+if [[ "$n" != crowbard2 && "$n" != crowbard3 && "$n" != "$crowbar_vmname" ]] ; then
```
Comment (jdsn) [inline hostscripts/gatehost/freshadminvm]: In double brackets please drop the quotes:
```
if [[ $n != crowbard2 && $n != crowbard3 && $n != $crowbar_vmname ]] ; then
```

File: hostscripts/gatehost/freshadminvm [hunk 4]
```diff
@@ -25,13 +25,15 @@ common_scripts="mkcloud-onhost.sh mkcloud-common.sh mkcloud-driver-$mkclouddrive
 for script in $common_scripts; do
     source ${scripts_lib_dir}/$script
 done
-admin_node_disk=/dev/system/$n
-[ -e $admin_node_disk ] || admin_node_disk=/dev/gate/$n
-[ -e $admin_node_disk ] || exit 51
+if [ -z "$admin_node_disk" ]; then
```
Comment (jdsn) [inline hostscripts/gatehost/freshadminvm]: Therefore use double brackets here:
```
if [[ ! $admin_node_disk ]]; then
```

File: hostscripts/gatehost/freshadminvm [hunk 5]
```diff
@@ -25,13 +25,15 @@ common_scripts="mkcloud-onhost.sh mkcloud-common.sh mkcloud-driver-$mkclouddrive
...
+        admin_node_disk=/dev/system/$n
+        [ -e $admin_node_disk ] || admin_node_disk=/dev/gate/$n
+        [ -e $admin_node_disk ] || exit 51
```
Comment (jdsn) [inline hostscripts/gatehost/freshadminvm]: Thats still insecure: no quotes in single brackets.
```
+ [[ -e $admin_node_disk ]] || admin_node_disk=/dev/gate/$n
+ [[ -e $admin_node_disk ]] || exit 51
```

File: hostscripts/gatehost/freshadminvm [hunk 6]
```diff
@@ -25,13 +25,15 @@ common_scripts="mkcloud-onhost.sh mkcloud-common.sh mkcloud-driver-$mkclouddrive
...
+fi
 onhost_prepareadmin
 virsh define $SCRIPTS_DIR/../hostscripts/gatehost/$n.xml # is OK to fail if already there
 virsh start $n
 sleep 100 # time for the admin VM to boot
-sshkey=`cat /home/jenkins/.ssh/id_rsa.pub`
+test -z "$sshkey" && sshkey=`cat /home/jenkins/.ssh/id_rsa.pub`
```
Comment (jdsn) [inline hostscripts/gatehost/freshadminvm]: either this, or shorter: `[[ ! $sshkey ]] && ...`

File: scripts/qa_crowbarsetup.sh (hunk 2 above)
Comment (jdsn) [reply scripts/qa_crowbarsetup.sh]: agreed, please move to the defaults section

but a nice idea indeed

File: hostscripts/gatehost/freshadminvm (hunk 5 above)
Comment (seife) [reply hostscripts/gatehost/freshadminvm]: $admin_node_diskis already checked for non-emptiness.

**Comment by nicolasbock** [comment]
@seife We typically rebase. Thanks!

File: hostscripts/gatehost/freshadminvm [hunk 7]
```diff
@@ -25,13 +25,15 @@ common_scripts="mkcloud-onhost.sh mkcloud-common.sh mkcloud-driver-$mkclouddrive
 for script in $common_scripts; do
     source ${scripts_lib_dir}/$script
 done
-admin_node_disk=/dev/system/$n
-[ -e $admin_node_disk ] || admin_node_disk=/dev/gate/$n
-[ -e $admin_node_disk ] || exit 51
+if [[ !-z $admin_node_disk ]]; then
```
Comment (bmwiedemann) [inline hostscripts/gatehost/freshadminvm]: the '!' should not be here - it is a syntax error and we do not want to negate here anyway.

File: hostscripts/gatehost/freshadminvm [hunk 8]
```diff
@@ -25,13 +25,15 @@ common_scripts="mkcloud-onhost.sh mkcloud-common.sh mkcloud-driver-$mkclouddrive
...
+        admin_node_disk=/dev/system/$n
+        [ -e $admin_node_disk ] || admin_node_disk=/dev/gate/$n
+        [ -e $admin_node_disk ] || exit 51
+fi
 onhost_prepareadmin
 virsh define $SCRIPTS_DIR/../hostscripts/gatehost/$n.xml # is OK to fail if already there
 virsh start $n
 sleep 100 # time for the admin VM to boot
-sshkey=`cat /home/jenkins/.ssh/id_rsa.pub`
+[[ ! $sshkey ]] && sshkey=`cat /home/jenkins/.ssh/id_rsa.pub`
```
Comment (bmwiedemann) [inline hostscripts/gatehost/freshadminvm]: `[[ $sshkey ]] || ...` might be easier to read?

File: hostscripts/gatehost/freshadminvm (hunk 7 above)
Comment (seife) [reply hostscripts/gatehost/freshadminvm]: well spotted ;-) fixing.

File: hostscripts/gatehost/freshadminvm (hunk 8 above)
Comment (seife) [reply hostscripts/gatehost/freshadminvm]: yes.

File: hostscripts/gatehost/freshadminvm (hunk 5 above)
Comment (nicolasbock) [reply hostscripts/gatehost/freshadminvm]: I would also prefer the `[[` style here. It's more consistent.

**Pull request merged**